pkg/api/controllers/volume_test.go

```diff
@@ -96,6 +98,64 @@ var (
 	fakeVolumes = []*model.VolumeSpec{fakeVolume}
 )
 
+func TestCreateVolume(t *testing.T) {
+	var jsonStr = []byte(`{
+		"id": "bd5b12a8-a101-11e7-941e-d77981b584d8",
+		"name":"fake Vol",
+		"description":"fake Vol",
+		"size": 1,
+		"availabilityZone": "default",
+		"profileId": "1106b972-66ef-11e7-b172-db03f3689c9c"
+	}`)
+
+	t.Run("Should return 202 if everything works well", func(t *testing.T) {
+		volume := model.VolumeSpec{BaseModel: &model.BaseModel{
+			Id: "bd5b12a8-a101-11e7-941e-d77981b584d8",
+			CreatedAt: time.Now().Format(constants.TimeFormat),
+		},
+		Size: int64(1),
+		AvailabilityZone: "default",
+		ProfileId: "1106b972-66ef-11e7-b172-db03f3689c9c",
+		Status: "creating",
+		}
+		json.NewDecoder(bytes.NewBuffer(jsonStr)).Decode(&volume)
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetDefaultProfile", c.NewAdminContext()).Return(&SampleProfiles[0], nil)
+		mockClient.On("GetProfile", c.NewAdminContext(), SampleVolumes[0].ProfileId).Return(&SampleProfiles[0], nil)
+		mockClient.On("CreateVolume", c.NewAdminContext(), &volume).Return(&SampleVolumes[0], nil)
+		db.C = mockClient
+
+		r, _ := http.NewRequest("POST", "/v1beta/block/volumes", bytes.NewBuffer(jsonStr))
+		w := httptest.NewRecorder()
+		r.Header.Set("Content-Type", "application/JSON")
+		beego.InsertFilter("*", beego.BeforeExec, func(httpCtx *context.Context) {
+			httpCtx.Input.SetData("context", c.NewAdminContext())
+		})
+		beego.BeeApp.Handlers.ServeHTTP(w, r)
+		var output model.VolumeSpec
+		json.Unmarshal(w.Body.Bytes(), &output)
+		assertTestResult(t, w.Code, 202)
+		assertTestResult(t, &output, &SampleVolumes[0])
+	})
+
+	t.Run("Should return 500 if create volume with bad request", func(t *testing.T) {
```

Comment:
why 500

Reply:
I will again update and check for bad request , may be we need to change main code as well

Reply:
Is this resolved?

Reply:
Resolved